base/io.jl

```diff
@@ -144,8 +144,8 @@ write(to::IO, p::Ptr) = write(to, convert(UInt, p))
 
 function write(s::IO, a::AbstractArray)
     nb = 0
-    for i in eachindex(a)
-        nb += write(s, a[i])
+    for b in a
+        nb += write(s, b)
```

Comment:
Would it be a little more readable to rename the input `A::AbstractArray` then do `for a in A` ? Having `nb` and `b` nearby is a little offputting (`nb` could be a few characters longer of course). Similar above with the `for y in x` comprehensions.